x-pack/solutions/security/plugins/security_solution/public/flyout/shared/utils/history_utils.ts

```diff
@@ -15,17 +16,28 @@ export const getProcessedHistory = ({
   history,
   maxCount,
 }: {
-  history: FlyoutPanelProps[];
+  history: FlyoutPanelHistory[];
   maxCount: number;
-}): FlyoutPanelProps[] => {
+}): FlyoutPanelHistory[] => {
   // Step 1: reverse history so the most recent is first
+  // We need to do this step first because we want to make sure that during step 2
+  // we are removing only older duplicates.
   const reversedHistory = history.slice().reverse();
 
   // Step 2: remove duplicates
-  const historyArray = Array.from(new Set(reversedHistory.map((i) => JSON.stringify(i)))).map((i) =>
-    JSON.parse(i)
-  );
+  // Because the lastOpen value will always be different, we're manually removing duplicates
+  // by looking at the panel's information only.
+  const uniqueHistory: FlyoutPanelHistory[] = [];
+  reversedHistory.forEach((hist) => {
+    const entryDoesNotExists =
+      uniqueHistory
+        .map((h: FlyoutPanelHistory) => JSON.stringify(h.panel))
```

Comment:
what do you think about using a set to check existence? Sets are generally faster when it comes to item look up. Something like
```
 const uniquePanels = new Set<string>();
 const uniqueHistory = reversedHistory.filter((hist) => {
    const panelString = JSON.stringify(hist.panel);
    const entryDoesNotExists = !uniquePanels.has(panelString);
    if (entryDoesNotExists) {
      uniquePanels.add(panelString);
      return true;
    }
    return false;
```

Reply:
yup good idea I will make that improvement!